# Orthofinder Results

Identifying "interesting" orthogroups and looking at their interproscan results

In [2]:
%matplotlib inline

In [3]:
import os
import matplotlib.pyplot as plt
import numpy as np
import pandas as pd

### Import Interproscan and Orthofinder results

In [4]:
interproscan_infn = "/home/jemimah/analysis/v3/interproscan/20200721_walli_all_protiens/Walli.v3.braker2.aa.both.tsv"
tsv_df = pd.read_csv(interproscan_infn, sep = '\t')

In [5]:
ortho = "/home/jemimah/analysis/v3/orthofinder/20200911/Results_Sep11/Orthogroups/Orthogroups.tsv"
ortho_species_df = pd.read_csv(ortho, sep = '\t')
ortho_species_df = ortho_species_df.set_index("Orthogroup")

ortho_count = "/home/jemimah/analysis/v3/orthofinder/20200911/Results_Sep11/Orthogroups/Orthogroups.GeneCount.tsv"
ortho_count_df = pd.read_csv(ortho_count, sep = '\t')
ortho_count_df = ortho_count_df.set_index("Orthogroup")

cols = ortho_count_df.columns.tolist()
ortho_species_df = ortho_species_df[cols[:-1]]

### Identify interesting results

So what I want to do is look through ortho_count_df for groups which are present in multiple species, but especially in *W ceracea* 

In [6]:
ortho_count_df.sort_values("W_ceracea", ascending=False ).iloc[0:120]

A_thaliana  S_lycopersicum  S_tuberosum  A_chinensis  D_carota  \
Orthogroup                                                                   
OG0000001            0               0            0            0         1   
OG0000004            0               0            0            1         0   
OG0000005            0               0            0            0         0   
OG0000006            0               3            1            0         2   
OG0000007            0               0            0            0         0   
...                ...             ...          ...          ...       ...   
OG0000125            3              11            6            4         8   
OG0002324            0               0            0            0         0   
OG0002070            0               0            1            0         0   
OG0002322            0               0            0            0         0   
OG0000025            9              20           43           19         1   

            L_sativa  H_annuus  W_ceracea  Total  
Orthogroup                                        
OG0000001          0         0        555    556  
OG0000004          0         2        440    443  
OG0000005          0         1        439    440  
OG0000006          0         1        374    381  
OG0000007          0         0        283    283  
...              ...       ...        ...    ...  
OG0000125          3        12         21     68  
OG0002324          0         0         21     21  
OG0002070          0         0         21     22  
OG0002322          0         0         21     21  
OG0000025          3         3         21    119  

[120 rows x 9 columns]

In [7]:
#"interesting" lists groups from
#ortho_count_df.sort_values("W_ceracea", ascending=False ).iloc[0:120]
#which contain genes from at least half of the non-W_ceracea species, preferably greater than 1,
#but with W ceracea having the greatest number.
#this does not apply to OG0000082 which has more H_annuus than W_ceracea
interesting = ["OG0000051", "OG0000107", "OG0000029", "OG0000130", "OG0000170", "OG0000208", "OG0000357",
               "OG0000339", "OG0000022", "OG0000062", "OG0000111", "OG0000647", "OG0000082", "OG0000019",
               "OG0000038", "OG0001315", "OG0000211", "OG0000190", "OG0000125"]


In [24]:
test_set = (ortho_count_df.sort_values("W_ceracea", ascending=False ).iloc[0:120])
test_set.loc[interesting]

A_thaliana  S_lycopersicum  S_tuberosum  A_chinensis  D_carota  \
Orthogroup                                                                   
OG0000051            1               0            1            4         0   
OG0000107            1               2            1            0         0   
OG0000029            0              18           26           16         0   
OG0000130            1               1            1            1         4   
OG0000170            1               3            2            5         3   
OG0000208            0               1            1            2         2   
OG0000357            1               3            0            1         0   
OG0000339            1               1            1            1         2   
OG0000022           19              16            8            2         9   
OG0000062            2               6           16            5         2   
OG0000111            0               8           10           18         2   
OG0000647            1               1            1            2         0   
OG0000082            0               0            0            0         1   
OG0000019            0              10           36           12        22   
OG0000038            6              12           13            8         5   
OG0001315            0               0            0            1         1   
OG0000211            0              10           13            0         0   
OG0000190            1              12           11            3         1   
OG0000125            3              11            6            4         8   

            L_sativa  H_annuus  W_ceracea  Total  
Orthogroup                                        
OG0000051          1         8         80     95  
OG0000107          0         6         62     72  
OG0000029          4         3         49    116  
OG0000130          2        16         41     67  
OG0000170          2         2         41     59  
OG0000208          6         4         38     54  
OG0000357          0         2         36     43  
OG0000339          1         1         36     44  
OG0000022         24        22         31    131  
OG0000062          6        23         30     90  
OG0000111          3         2         28     71  
OG0000647          0         0         28     33  
OG0000082         18        35         26     80  
OG0000019         15        20         25    140  
OG0000038         10        24         25    103  
OG0001315          1         1         22     26  
OG0000211          5         4         22     54  
OG0000190          4         3         21     56  
OG0000125          3        12         21     68

In [25]:
ortho_species_df.loc[interesting]

A_thaliana  \
Orthogroup                                                      
OG0000051                                         AT4G13320.1   
OG0000107                                         ATMG00750.1   
OG0000029                                                 NaN   
OG0000130                                         AT1G45616.1   
OG0000170                                         AT2G45400.1   
OG0000208                                                 NaN   
OG0000357                                         AT3G29785.1   
OG0000339                                         AT5G08490.3   
OG0000022   AT1G33540.1, AT1G73270.1, AT1G73280.1, AT1G732...   
OG0000062                            AT1G12615.1, AT1G12700.2   
OG0000111                                                 NaN   
OG0000647                                         AT5G36930.2   
OG0000082                                                 NaN   
OG0000019                                                 NaN   
OG0000038   AT2G17680.1, AT4G35660.1, AT4G35680.1, AT4G356...   
OG0001315                                                 NaN   
OG0000211                                                 NaN   
OG0000190                                         AT4G27290.2   
OG0000125               AT1G15940.1, AT1G80810.3, AT4G31880.1   

                                               S_lycopersicum  \
Orthogroup                                                      
OG0000051                                                 NaN   
OG0000107              Solyc07g032120.1.1, Solyc10g045120.1.1   
OG0000029   Solyc02g036270.2.1, Solyc03g094100.1.1, Solyc0...   
OG0000130                                  Solyc12g013730.1.1   
OG0000170   Solyc01g094070.2.1, Solyc05g051010.2.1, Solyc0...   
OG0000208                                  Solyc04g078410.2.1   
OG0000357   Solyc00g005160.1.1, Solyc03g096270.1.1, Solyc0...   
OG0000339                                  Solyc10g005090.2.1   
OG0000022   Solyc04g076120.2.1, Solyc04g077630.2.1, Solyc0...   
OG0000062   Solyc06g005220.2.1, Solyc06g007280.2.1, Solyc0...   
OG0000111   Solyc01g102840.2.1, Solyc01g102850.1.1, Solyc0...   
OG0000647                                  Solyc09g007710.2.1   
OG0000082                                                 NaN   
OG0000019   Solyc02g079610.2.1, Solyc02g081470.2.1, Solyc0...   
OG0000038   Solyc02g082980.1.1, Solyc02g083000.1.1, Solyc0...   
OG0001315                                                 NaN   
OG0000211   Solyc04g051880.1.1, Solyc04g072490.2.1, Solyc0...   
OG0000190   Solyc07g063700.1.1, Solyc07g063710.1.1, Solyc0...   
OG0000125   Solyc02g086630.1.1, Solyc02g092850.1.1, Solyc0...   

                                                  S_tuberosum  \
Orthogroup                                                      
OG0000051                                PGSC0003DMP400047901   
OG0000107                                PGSC0003DMP400021359   
OG0000029   PGSC0003DMP400001764, PGSC0003DMP400005107, PG...   
OG0000130                                PGSC0003DMP400038082   
OG0000170          PGSC0003DMP400012427, PGSC0003DMP400041037   
OG0000208                                PGSC0003DMP400005946   
OG0000357                                                 NaN   
OG0000339                                PGSC0003DMP400019952   
OG0000022   PGSC0003DMP400014101, PGSC0003DMP400018772, PG...   
OG0000062   PGSC0003DMP400006067, PGSC0003DMP400006070, PG...   
OG0000111   PGSC0003DMP400005063, PGSC0003DMP400022408, PG...   
OG0000647                                PGSC0003DMP400048338   
OG0000082                                                 NaN   
OG0000019   PGSC0003DMP400008038, PGSC0003DMP400010170, PG...   
OG0000038   PGSC0003DMP400002767, PGSC0003DMP400002768, PG...   
OG0001315                                                 NaN   
OG0000211   PGSC0003DMP400011178, PGSC0003DMP400013875, PG...   
OG0000190   PGSC0003DMP400033420, PGSC0003DMP400033421, PG...   
OG0000125   PGSC000

In [30]:
def ips_summary(orthogroup_list, disp="all"):
    for x in orthogroup_list:
        ortho_wlist = ortho_species_df.loc[x]["W_ceracea"].split(", ")
        ortho_all = tsv_df[tsv_df["Protein_accession"].isin(ortho_wlist)][["Protein_accession", "Contig", "SequenceMD5", "Analysis", "Signature_accession","Signature_description", "Interpro_annotation", "Interpro_description"]]
        ortho_unique = ortho_all.drop_duplicates(subset="Signature_accession")[["Analysis", "Signature_accession","Signature_description", "Interpro_annotation", "Interpro_description"]]
        counts = ortho_all.pivot_table(index=['Signature_accession'], aggfunc='size')
        ortho_unique["count"] = ortho_unique["Signature_accession"].map(counts.to_dict())
        ortho_unique = ortho_unique.sort_values("count", ascending = False )
        ortho_unique
#        print(x, "  \nWalli genes:", len(ortho_wlist), "  \nAll interpro results:", len(ortho_all),
#             "  \nUnique interpro results:", len(ortho_unique))
#for notebook copy-paste
        if disp == "summary":
            print("**" + x + "**  \nWalli genes:", len(ortho_wlist), "  \nAll interpro results:", len(ortho_all),
             "  \nUnique interpro results:", len(ortho_unique), "  ")
        elif disp == "table":
            print(x)
            display(HTML(ortho_unique.to_html()))
        elif disp == "all":
            print("**" + x + "**  \nWalli genes:", len(ortho_wlist), "  \nAll interpro results:", len(ortho_all),
             "  \nUnique interpro results:", len(ortho_unique), "  ")
            display(HTML(ortho_unique.to_html()))
        else:
            print("Please declare summary, table, or all")

**All interesting results:**

In [31]:
#ips_summary(interesting, "summary")

Interpret results:
- PANTHER accessions can be searched here http://www.pantherdb.org/panther/
- Gene3D: add the numbers to the end of this link http://www.cathdb.info/version/v4_2_0/superfamily/ eg http://www.cathdb.info/version/v4_2_0/superfamily/2.60.120.330
- Superfamily http://supfam.org/SUPERFAMILY/
- I generally just google the rest.

In [32]:
ips_summary(interesting, "all")

**OG0000051**  
Walli genes: 80   
All interpro results: 995   
Unique interpro results: 55   


**OG0000107**  
Walli genes: 62   
All interpro results: 660   
Unique interpro results: 33   


**OG0000029**  
Walli genes: 49   
All interpro results: 391   
Unique interpro results: 21   


**OG0000130**  
Walli genes: 41   
All interpro results: 758   
Unique interpro results: 31   


**OG0000170**  
Walli genes: 41   
All interpro results: 243   
Unique interpro results: 10   


**OG0000208**  
Walli genes: 38   
All interpro results: 571   
Unique interpro results: 38   


**OG0000357**  
Walli genes: 36   
All interpro results: 303   
Unique interpro results: 20   


**OG0000339**  
Walli genes: 36   
All interpro results: 806   
Unique interpro results: 6   


**OG0000022**  
Walli genes: 31   
All interpro results: 302   
Unique interpro results: 11   


**OG0000062**  
Walli genes: 30   
All interpro results: 936   
Unique interpro results: 24   


**OG0000111**  
Walli genes: 28   
All interpro results: 460   
Unique interpro results: 23   


**OG0000647**  
Walli genes: 28   
All interpro results: 388   
Unique interpro results: 27   


**OG0000082**  
Walli genes: 26   
All interpro results: 157   
Unique interpro results: 6   


**OG0000019**  
Walli genes: 25   
All interpro results: 244   
Unique interpro results: 20   


**OG0000038**  
Walli genes: 25   
All interpro results: 93   
Unique interpro results: 7   


**OG0001315**  
Walli genes: 22   
All interpro results: 144   
Unique interpro results: 18   


**OG0000211**  
Walli genes: 22   
All interpro results: 186   
Unique interpro results: 6   


**OG0000190**  
Walli genes: 21   
All interpro results: 502   
Unique interpro results: 31   


**OG0000125**  
Walli genes: 21   
All interpro results: 123   
Unique interpro results: 8   


## Protien Domains
**OG0000051** - seems likely to be TE  
Walli genes: 80   
All interpro results: 995   
Unique interpro results: 55  
- two most popular Panther results are "viral or transposable element protein"
- viral keeps coming up (as well as the ones that say it there, http://www.cathdb.info/version/latest/superfamily/3.30.70.270 for Gene3D)  

**OG0000107** - seems likely to be TE  
Walli genes: 62   
All interpro results: 660   
Unique interpro results: 33  
- same reasons as the one above 

**OG0000029** - disease resistance  
Walli genes: 49   
All interpro results: 391   
Unique interpro results: 21  
- Prints result is for disease resistence (hard to find more info on, though)
- PTHR33463:SF21 can't be found??? but PTHR33463 is a family with 973 genes, some of the subfamilies being disease resistence http://www.pantherdb.org/list/list.do?numPerPage=100&save=yes&searchModType=numperpage&listType=6

**OG0000130** - Strong support for a Leucine-rich repeat, but that doesnt seem to be informative in and of itself  
Walli genes: 41   
All interpro results: 758   
Unique interpro results: 31  
- panther results http://www.pantherdb.org/panther/family.do?clsAccession=PTHR27004:SF112,  http://www.pantherdb.org/genes/geneList.do?searchType=basic&fieldName=all&listType=1&fieldValue=PTHR27004&organism=all are receptors and/or leucine rich http://www.pantherdb.org/panther/familyList.do?searchType=basic&fieldName=all&organism=all&listType=6&fieldValue=PTHR46662

**OG0000170** - NAD dependent something  
Walli genes: 41   
All interpro results: 243   
Unique interpro results: 10  
- Gene3D: http://www.cathdb.info/version/latest/superfamily/3.40.50.720
- Panther: http://www.pantherdb.org/panther/family.do?clsAccession=PTHR10366:SF563
- CDD: https://www.ncbi.nlm.nih.gov/Structure/cdd/cddsrv.cgi?uid=cd08958 - appears to act in the reduction of flavinoids (more info in link)

**OG0000208** - Protein Kinases  
Walli genes: 38   
All interpro results: 571   
Unique interpro results: 38  
- Panther is also Protein Kinases

**OG0000357** -   Really uncertain
Walli genes: 36   
All interpro results: 303   
Unique interpro results: 20   
- Panther is Zinc finger

**OG0000339** - Pentatricopeptide repeat  
Walli genes: 36   
All interpro results: 806   
Unique interpro results: 6  
- RNA level gene expression regulation (https://www.sciencedirect.com/science/article/pii/S030090841500108X)  

**OG0000022** - Peptidase S10, serine carboxypeptidase   
Walli genes: 31   
All interpro results: 302   
Unique interpro results: 11  
- Calalytic something or other? (https://www.ebi.ac.uk/interpro/entry/InterPro/IPR001563/)

**OG0000062** - Pentatricopeptide (PPR) repeat profile  
Walli genes: 30    
All interpro results: 936   
Unique interpro results: 24  
- "Most of PPR proteins have roles in mitochondria or plastid" (https://prosite.expasy.org/PDOC51375)  

**OG0000111** - Maybe disease resistance?   
Walli genes: 28    
All interpro results: 460   
Unique interpro results: 23  
- Seems to be a mixture of results I'vve already had?
    - PRINTS reports desiease resistence
    - Gene3D. Panther, others report Leucine-rich repeat domain
    - Toll/interleukin results seem to be anti bacterial/fungal (incl in plant cytoplasmic host defence https://www.ebi.ac.uk/interpro/entry/InterPro/IPR000157/)

**OG0000647** - very simillar to the one above   
Walli genes: 28   
All interpro results: 388   
Unique interpro results: 27  
- might actuaklly be the same results (with different counts)
- if they're so similalr, why are they two different orthogroups?

**OG0000082** -   
Walli genes: 26   
All interpro results: 157   
Unique interpro results: 6  
- Gene3D appears to be antibiotic resistence? which isn't useful surely (https://www.ebi.ac.uk/interpro/entry/InterPro/IPR023213/)
- seems to be a big pfam family: https://pfam.xfam.org/family/Transferase
- Panther: http://www.pantherdb.org/panther/family.do?clsAccession=PTHR31623:SF18

**OG0000019** -  Protein kinases, maybe rust resistent?
Walli genes: 25   
All interpro results: 244   
Unique interpro results: 20  
- http://www.pantherdb.org/panther/family.do?clsAccession=PTHR27009: RUST RESISTANCE KINASE LR10-RELATED
    - cannot find how it is rust resistant??
    
**OG0000038** - Very uncertain    
Walli genes: 25   
All interpro results: 93   
Unique interpro results: 7  
- Panther has it as a BPS1-LIKE PROTEIN (http://www.pantherdb.org/panther/family.do?clsAccession=PTHR31509), which is described here: https://www.ncbi.nlm.nih.gov/gene?Db=gene&Cmd=DetailsSearch&Term=839536 as being inhibiting of auxing signalling
- and also as this other unknown rice gene? http://www.pantherdb.org/panther/family.do?clsAccession=PTHR33070; https://www.uniprot.org/uniprot/Q5Z978

**OG0001315** - interacts with other protiens?   
Walli genes: 22   
All interpro results: 144   
Unique interpro results: 18  
- F-box domain and Leucine rich domain
    - "F-box domains commonly exist in proteins in concert with other protein–protein interaction motifs such as leucine-rich repeats" at least my result matches wikipedia (https://en.wikipedia.org/wiki/F-box_protein)?
    - so seems to suggest protien-protien interactions, although of an unknown sort
- idk if the FAD binding domain matches are useful because they are pretty infrequent

**OG0000211** - Mitochondira/chloroplast   
Walli genes: 22   
All interpro results: 186   
Unique interpro results: 6  
- seems smooth match? 
- although the panther result just seems to be unknown: http://www.pantherdb.org/panther/family.do?clsAccession=PTHR13068

**OG0000190** -    
Walli genes: 21   
All interpro results: 502   
Unique interpro results: 31  
- Panther domain is unclear
- I really dont understand what the pan/apple domain is: https://www.ebi.ac.uk/interpro/entry/InterPro/IPR003609/
- for the bulb-lectin one: https://prosite.expasy.org/doc/PS50927
- the fact that there is almost exactly the same count of a bunch of different things makes me sure they are all realated results. I just dont know what they're meaning.

**OG0000125** -   
Walli genes: 21  
All interpro results: 123   
Unique interpro results: 8  
- Panther result suggets DNA repair and chromatin-chromatin binding (http://www.pantherdb.org/panther/family.do?clsAccession=PTHR12663)
- the TUDOR/PWWP/MBT SUPERFAMILY PROTEIN stuff also seems to be about chromatid cohesion (https://www.uniprot.org/uniprot/A8MRD9-1 - couldnt find the exact right one)
- have't managed to figure out what an ARM repeat is (http://supfam.org/SUPERFAMILY/cgi-bin/scop.cgi?ipid=SSF48371)

### Checking for more

Looking to see if there are any more interesting results I haven't seen yet

In [33]:
ortho_count_df["prop"] = ortho_count_df["W_ceracea"]/ortho_count_df["Total"]

In [34]:
ortho_count_df.loc[interesting].sort_values("prop", ascending=False )

A_thaliana  S_lycopersicum  S_tuberosum  A_chinensis  D_carota  \
Orthogroup                                                                   
OG0000107            1               2            1            0         0   
OG0000647            1               1            1            2         0   
OG0001315            0               0            0            1         1   
OG0000051            1               0            1            4         0   
OG0000357            1               3            0            1         0   
OG0000339            1               1            1            1         2   
OG0000208            0               1            1            2         2   
OG0000170            1               3            2            5         3   
OG0000130            1               1            1            1         4   
OG0000029            0              18           26           16         0   
OG0000211            0              10           13            0         0   
OG0000111            0               8           10           18         2   
OG0000190            1              12           11            3         1   
OG0000062            2               6           16            5         2   
OG0000082            0               0            0            0         1   
OG0000125            3              11            6            4         8   
OG0000038            6              12           13            8         5   
OG0000022           19              16            8            2         9   
OG0000019            0              10           36           12        22   

            L_sativa  H_annuus  W_ceracea  Total      prop  
Orthogroup                                                  
OG0000107          0         6         62     72  0.861111  
OG0000647          0         0         28     33  0.848485  
OG0001315          1         1         22     26  0.846154  
OG0000051          1         8         80     95  0.842105  
OG0000357          0         2         36     43  0.837209  
OG0000339          1         1         36     44  0.818182  
OG0000208          6         4         38     54  0.703704  
OG0000170          2         2         41     59  0.694915  
OG0000130          2        16         41     67  0.611940  
OG0000029          4         3         49    116  0.422414  
OG0000211          5         4         22     54  0.407407  
OG0000111          3         2         28     71  0.394366  
OG0000190          4         3         21     56  0.375000  
OG0000062          6        23         30     90  0.333333  
OG0000082         18        35         26     80  0.325000  
OG0000125          3        12         21     68  0.308824  
OG0000038         10        24         25    103  0.242718  
OG0000022         24        22         31    131  0.236641  
OG0000019         15        20         25    140  0.178571

In [35]:
not_yet_df = ortho_count_df[~ortho_count_df.index.isin(interesting)]

In [36]:
not_yet_df[not_yet_df["prop"] != 1.0].sort_values("prop", ascending=False).iloc[420:450]

A_thaliana  S_lycopersicum  S_tuberosum  A_chinensis  D_carota  \
Orthogroup                                                                   
OG0022217            0               0            1            0         0   
OG0020384            0               0            0            0         1   
OG0022219            0               0            1            0         0   
OG0022221            0               0            1            0         0   
OG0019983            0               0            0            1         0   
OG0016768            0               1            1            0         0   
OG0000724            1               1            1            2         1   
OG0003558            2               0            0            2         1   
OG0003647            0               1            1            1         1   
OG0003644            0               0            0            1         3   
OG0000769            0               3            2            1         5   
OG0005358            1               1            1            2         0   
OG0005510            0               0            2            1         0   
OG0001077            0               4            2            2         0   
OG0008911            0               2            0            0         2   
OG0002058            0               0            3            0         0   
OG0008264            0               0            0            2         0   
OG0002833            1               1            1            1         1   
OG0002836            0               1            1            1         1   
OG0013890            0               1            1            1         0   
OG0014489            0               0            0            0         0   
OG0014490            0               0            0            0         0   
OG0014155            1               0            0            1         0   
OG0013315            0               0            0            1         1   
OG0014322            1               0            0            0         0   
OG0013975            0               1            1            1         0   
OG0013540            0               0            0            1         0   
OG0014406            0               0            0            0         3   
OG0014407            0               1            0            0         1   
OG0014443            0               0            2            0         0   

            L_sativa  H_annuus  W_ceracea  Total      prop  
Orthogroup                                                  
OG0022217          0         0          2      3  0.666667  
OG0020384          0         0          2      3  0.666667  
OG0022219          0         0          2      3  0.666667  
OG0022221          0         0          2      3  0.666667  
OG0019983          0         0          2      3  0.666667  
OG0016768          0         0          4      6  0.666667  
OG0000724          3         2         21     32  0.656250  
OG0003558          0         1         11     17  0.647059  
OG0003647          1         1         11     17  0.647059  
OG0003644          1         1         11     17  0.647059  
OG0000769          0         0         20     31  0.645161  
OG0005358          0         0          9     14  0.642857  
OG0005510          0         2          9     14  0.642857  
OG0001077          2         0         18     28  0.642857  
OG0008911          0         0          7     11  0.636364  
OG0002058          0         5         14     22  0.636364  
OG0008264          1         1          7     11  0.636364  
OG0002833          1         1         12     19  0.631579  
OG0002836          2         1         12     19  0.631579  
OG0013890          0         0          5      8  0.625000  
OG0014489          2         1          5      8  0.625000  
OG0014490          0         3          5      8  0.625000  
OG0014155          0         1          5      8  0

In [37]:
#"interesting_2" lists groups from
#not_yet_df[not_yet_df["prop"] != 1.0].sort_values("prop", ascending=False)
#where not_yet_df = ortho_count_df[~ortho_count_df.index.isin(interesting)]
# which contain genes from at least half of the non-W_ceracea species,
# with W ceracea having a propotion of the total genes > 0.625
interesting_2 = ["OG0001651", "OG0004401", "OG0003049", "OG0001024", "OG0002516", "OG0001842", 
                 "OG0001075", "OG0002249", "OG0003266", "OG0004831", "OG0000724", "OG0003558", 
                 "OG0003647", "OG0003644", "OG0000769", "OG0005358", "OG0001077", "OG0002833", 
                 "OG0002836"]

In [38]:
ips_summary(interesting_2, "all")

**OG0001651**  
Walli genes: 19   
All interpro results: 170   
Unique interpro results: 21   


**OG0004401**  
Walli genes: 11   
All interpro results: 58   
Unique interpro results: 13   


**OG0003049**  
Walli genes: 13   
All interpro results: 79   
Unique interpro results: 9   


**OG0001024**  
Walli genes: 20   
All interpro results: 100   
Unique interpro results: 14   


**OG0002516**  
Walli genes: 14   
All interpro results: 76   
Unique interpro results: 11   


**OG0001842**  
Walli genes: 16   
All interpro results: 95   
Unique interpro results: 12   


**OG0001075**  
Walli genes: 19   
All interpro results: 52   
Unique interpro results: 3   


**OG0002249**  
Walli genes: 14   
All interpro results: 26   
Unique interpro results: 10   


**OG0003266**  
Walli genes: 12   
All interpro results: 99   
Unique interpro results: 8   


**OG0004831**  
Walli genes: 10   
All interpro results: 124   
Unique interpro results: 8   


**OG0000724**  
Walli genes: 21   
All interpro results: 147   
Unique interpro results: 10   


**OG0003558**  
Walli genes: 11   
All interpro results: 64   
Unique interpro results: 12   


**OG0003647**  
Walli genes: 11   
All interpro results: 100   
Unique interpro results: 6   


**OG0003644**  
Walli genes: 11   
All interpro results: 20   
Unique interpro results: 3   


**OG0000769**  
Walli genes: 20   
All interpro results: 49   
Unique interpro results: 3   


**OG0005358**  
Walli genes: 9   
All interpro results: 116   
Unique interpro results: 11   


**OG0001077**  
Walli genes: 18   
All interpro results: 117   
Unique interpro results: 6   


**OG0002833**  
Walli genes: 12   
All interpro results: 99   
Unique interpro results: 20   


**OG0002836**  
Walli genes: 12   
All interpro results: 24   
Unique interpro results: 3   


## Protien Domains (second set)

**OG0001651** - Ankyrin repeat: protein-protein interaction motif
Walli genes: 19   
All interpro results: 170   
Unique interpro results: 21   
- Ankyrin repeat-containing domain is a popularconclusion (https://www.ebi.ac.uk/interpro/entry/InterPro/IPR020683/)
- with a couple of retrotransposon matches

**OG0004401**  
Walli genes: 11   
All interpro results: 58   
Unique interpro results: 13   
- SUPERFAMILY and Gene3D have P-loop containing nucleoside triphosphate hydrolase: https://www.ebi.ac.uk/interpro/entry/InterPro/IPR027417/ ; http://www.cathdb.info/version/v4_2_0/superfamily/3.40.50.300
- Panther abd Pfam has it as an transporter, though when you look at the GO terms from panther they also have it as a presponse to stress
- then at a second level a lot of them have Citrate Synthase

**OG0003049**  
Walli genes: 13   
All interpro results: 79   
Unique interpro results: 9   
- Gene3D has it as an antibiotic? http://www.cathdb.info/version/v4_2_0/superfamily/2.60.120.330
- but all the rest seem to just be 2OG-Fe(II) oxygenase superfamily	- apparently the families overlap? http://www.ebi.ac.uk/interpro/entry/InterPro/IPR026992/

**OG0001024**  
Walli genes: 20   
All interpro results: 100   
Unique interpro results: 14   
- Seems very definitely a GOLGIN, which sems to be about maintaining the golgi apparatus https://www.uniprot.org/uniprot/Q7SXE4
- apparently that explains all the coils? https://en.wikipedia.org/wiki/GOLGA5

**OG0002516** - Probably a TE?  
Walli genes: 14   
All interpro results: 76   
Unique interpro results: 11   
- Panther: SERINE/THREONINE-PROTEIN PHOSPHATASE 7 LONG FORM-LIKE PROTEIN (PTHR46033:SF16) http://www.pantherdb.org/panther/family.do?clsAccession=PTHR46033:SF16
    - interestingly the lable they have in the signature description belongs to PTHR46033:SF20 http://www.pantherdb.org/panther/family.do?clsAccession=PTHR46033:SF20 and is a TE
- pfam has it as transposable http://pfam.xfam.org/family/PF10536

**OG0001842** - Lipase  
Walli genes: 16   
All interpro results: 95   
Unique interpro results: 12   
- Panther: GDSL LIPASE/ACYLHYDROLASE FAMILY PROTEIN (AFU_ORTHOLOGUE AFUA_4G14700) (PTHR45648)

**OG0001075**  
Walli genes: 19   
All interpro results: 52   
Unique interpro results: 3   
- MYOSIN-LIKE PROTEIN (PTHR35468): Myosins are motor protiens that make up muscles in animals.

**OG0002249**  
Walli genes: 14   
All interpro results: 26   
Unique interpro results: 10  
- appears to be a V-ATPase (proton pump) or an armadillo fold

**OG0003266** - Aspartic peptidase  
Walli genes: 12   
All interpro results: 99   
Unique interpro results: 8   
- aspartic peptidase https://www.ebi.ac.uk/interpro/entry/InterPro/IPR021109/
- even the Xylanase inhibitors are also hound in the aspartic peptidases: https://www.ebi.ac.uk/interpro/entry/InterPro/IPR032861/

**OG0004831** - Pentatricopeptide repeat  
Walli genes: 10   
All interpro results: 124   
Unique interpro results: 8   
- See OG0000062 and OG0000339

**OG0000724** - Calcium/lipid-binding domain, CaLB  
Walli genes: 21   
All interpro results: 147   
Unique interpro results: 10   
- https://www.ncbi.nlm.nih.gov/gene?cmd=DetailsSearch&term=AT3G61050: "It can bind ceramide and is involved in drought and salt tolerance."

**OG0003558**  
Walli genes: 11   
All interpro results: 64   
Unique interpro results: 12   
- Glycosidases and Glycosyl hydrolase are the same. Can't figure out what they do though
- FRIGIDA-like protein 1 in arabidopsis: https://www.uniprot.org/uniprot/Q9FFF1
     - Panther is also frigida-like
     
**OG0003647** - Pentatricopeptide repeat  
Walli genes: 11   
All interpro results: 100   
Unique interpro results: 6   
- There are a few more up there

**OG0003644**  
Walli genes: 11   
All interpro results: 20   
Unique interpro results: 3   
- un-named panther family

**OG0000769**  
Walli genes: 20   
All interpro results: 49   
Unique interpro results: 3   
- HOMEODOMAIN PROTEIN 8, PUTATIVE-RELATED (http://www.pantherdb.org/panther/family.do?clsAccession=PTHR35121:SF4)
- probably a transcription factor https://ghr.nlm.nih.gov/primer/genefamily/homeoboxes

**OG0005358**  
Walli genes: 9   
All interpro results: 116   
Unique interpro results: 11   
- Panther is the same as the ProSiteProfiles one below it
- Seems to be a DNA binding something: B3 is exclusively TFs https://en.wikipedia.org/wiki/B3_domain

**OG0001077** - Glycosyltransferase/glycogen phosphorylase - anti-biotics  
Walli genes: 18   
All interpro results: 117   
Unique interpro results: 6  
- cdd link https://www.ncbi.nlm.nih.gov/Structure/cdd/cd03784 - involved in the final stages of the biosynthesis of antibiotics

**OG0002833** - Cyclin-like & Leucine rich  
Walli genes: 12   
All interpro results: 99   
Unique interpro results: 20   
- gene3D http://www.cathdb.info/version/v4_2_0/superfamily/1.10.472.10

**OG0002836** - unclear  
Walli genes: 12   
All interpro results: 24   
Unique interpro results: 3  
-Panther http://www.pantherdb.org/panther/family.do?clsAccession=PTHR36485:SF1, but function unclear

So I would prabably like to look further into:
- The ones for TFs
- The  Pentatricopeptide repeats: appear to be mitochondria interacting? https://en.wikipedia.org/wiki/Pentatricopeptide_repeat
- The disease resistance ones

And should re-check both lists to see if anything else looks worth it

In [39]:
#print("\n".join(interesting + interesting_2))

## Summary of 38 "interesting" orthogroups
TEs:
- OG0000051
- OG0000107
- OG0002516
- OG0000357 (probably)

Disease?:
- OG0000029
- OG0000111 - a bit complicated
- OG0000647 - a bit complicated
- OG0003049 - antibiotic?
- OG0001077 - antibiotic?

Protien Kinase:
- OG0000208
- OG0000019

Pentatricopeptide repeat (mitochondria interacting):
- OG0000339
- OG0000062
- OG0004831
- OG0003647

Mitochondria/chloroplast:
- OG0000211

Protien interactions:
- OG0001315
- OG0001651

Possibly a Transcription factor:
- OG0000769
- OG0005358

Odd but specific:
- OG0000130 - Leucine rich repeat
- OG0002833 - Leucine rich & Cyclin-like
- OG0000170 - NAD-dependent something
- OG0000022 - Peptidase S10, serine carboxypeptidase
- OG0001024 - Golgi
- OG0001842 - Lipase
- OG0001075 - Myosin-like
- OG0000125 - Chromatin
- OG0002249 - V-ATPase (Proton pump)
- OG0003266 - Aspartic peptidase  
- OG0000724 - Calcium/lipid-binding domain, CaLB: salt and drought tolerance 
- OG0000082 - transferase of some sort

Unclear:
- OG0000038 - a few possible results (auxin signalling inhibition, other unknown genes)
- OG0000190 - a lot of info but idk how its all related
- OG0004401 - a few possible results (transporter, Citrate Synthase)
- OG0003558 - Glycosyl hydrolase and/or FRIGIDA-like protein 
- OG0003644 - no info
- OG0002836 - unclear




In [40]:
counts_interested = ortho_count_df.loc[(interesting + interesting_2)] 

In [41]:
dict_conc = {"OG0000051": "TEs", "OG0000107": "TEs",
             "OG0002516": "TEs", "OG0000357": "TEs", 
             "OG0000029": "Disease", "OG0000111": "Disease", 
             "OG0000647": "Disease", "OG0003049": "Disease - antibiotic?", 
             "OG0001077":  "Disease - antibiotic?",
             "OG0000208": "Protien Kinase", "OG0000019": "Protien Kinase",
             "OG0000339": "Pentatricopeptide repeat", "OG0000062": "Pentatricopeptide repeat", 
             "OG0004831": "Pentatricopeptide repeat", "OG0003647": "Pentatricopeptide repeat",
             "OG0000211": "Mitochondria/chloroplast",
             "OG0001315": "Protien interactions", "OG0001651": "Protien interactions",
             "OG0000769": "transcription factor possibly", "OG0005358": "transcription factor possibly",
             "OG0000130": "Leucine rich repeat", "OG0000170": "NAD-dependent", 
             "OG0000022": "Peptidase S10, serine carboxypeptidase", "OG0001024": "Golgi",
             "OG0001842": "Lipase", "OG0001075": "Myosin-like",
             "OG0000125": "Chromatin", "OG0002249": "V-ATPase (Proton pump)",
             "OG0003266": "Aspartic peptidase", "OG0000724": "Calcium/lipid-binding domain, CaLB: salt and drought tolerance",
             "OG0000082": "transferase of some sort", "OG0002833": "Leucine rich & Cyclin-like",
             "OG0000038": "Unclear",
             "OG0000190": "Unclear", "OG0004401": "Unclear",
             "OG0003558": "Unclear", "OG0002836": "Unclear",
             "OG0003644": "no info"}


In [42]:
counts_interested["info"] = counts_interested.index.map(dict_conc)

## Summary Table

In [43]:
counts_interested.sort_values("info")

A_thaliana  S_lycopersicum  S_tuberosum  A_chinensis  D_carota  \
Orthogroup                                                                   
OG0003266            0               1            1            1         1   
OG0000724            1               1            1            2         1   
OG0000125            3              11            6            4         8   
OG0000111            0               8           10           18         2   
OG0000647            1               1            1            2         0   
OG0000029            0              18           26           16         0   
OG0003049            0               0            1            2         0   
OG0001077            0               4            2            2         0   
OG0001024            1               1            0            2         1   
OG0002833            1               1            1            1         1   
OG0000130            1               1            1            1         4   
OG0001842            0               1            1            1         1   
OG0000211            0              10           13            0         0   
OG0001075            1               1            1            2         1   
OG0000170            1               3            2            5         3   
OG0004831            0               1            1            1         0   
OG0000062            2               6           16            5         2   
OG0000339            1               1            1            1         2   
OG0003647            0               1            1            1         1   
OG0000022           19              16            8            2         9   
OG0000019            0              10           36           12        22   
OG0000208            0               1            1            2         2   
OG0001651            0               1            1            0         0   
OG0001315            0               0            0            1         1   
OG0000051            1               0            1            4         0   
OG0000357            1               3            0            1         0   
OG0000107            1               2            1            0         0   
OG0002516            0               0            2            1         1   
OG0004401            0               0            1            1         1   
OG0000190            1              12           11            3         1   
OG0000038            6              12           13            8         5   
OG0003558            2               0            0            2         1   
OG0002836            0               1            1            1         1   
OG0002249            1               1            0            2         1   
OG0003644            0               0            0            1         3   
OG0000769            0               3            2            1         5   
OG0005358            1               1            1            2         0   
OG0000082            0               0            0            0         1   

            L_sativa  H_annuus  W_ceracea  Total      prop  \
Orthogroup                                                   
OG0003266          1         1         12     18  0.666667   
OG0000724          3         2         21     32  0.656250   
OG0000125          3        12         21     68  0.308824   
OG0000111          3         2         28     71  0.394366   
OG0000647          0         0         28     33  0.848485   
OG0000029          4         3         49    116  0.422414   
OG0003049          1         1         13     18  0.722222   
OG0001077          2         0         18     28  0.642857   
OG0001024          1         2         20     28  0.714286   
OG0002833          1         1         12     19  0.631579   
OG0000130          2        16         41     67  0.611940   
OG0001842          2         1         16     23  0.695652   
OG0000211          5   

In [44]:
#(counts_interested.sort_values("info")).to_csv("orthogroups_interesting.tsv", sep="\t")

### Further research

From Ben: "For the OGs of interest I would pick 3-4 groups and look what orthologs in Arabidopsis or so do if available. Go for the transcription factors, protein kinase, disease and maybe Calcium stuff. If no Arabidopsis available just do a blast search and see if any other homolog has a described function so you can write a bit of a story."

In [45]:
#counts_interested["info"].unique()

In [46]:
TFs = counts_interested[counts_interested["info"] == 'transcription factor possibly']
pk = counts_interested[counts_interested["info"] == 'Protien Kinase']
dis = counts_interested[counts_interested["info"].str.contains("Disease")]
cal = counts_interested[counts_interested["info"] == 'Calcium/lipid-binding domain, CaLB: salt and drought tolerance']

In [47]:
TFs.append(pk).append(dis).append(cal)

A_thaliana  S_lycopersicum  S_tuberosum  A_chinensis  D_carota  \
Orthogroup                                                                   
OG0000769            0               3            2            1         5   
OG0005358            1               1            1            2         0   
OG0000208            0               1            1            2         2   
OG0000019            0              10           36           12        22   
OG0000029            0              18           26           16         0   
OG0000111            0               8           10           18         2   
OG0000647            1               1            1            2         0   
OG0003049            0               0            1            2         0   
OG0001077            0               4            2            2         0   
OG0000724            1               1            1            2         1   

            L_sativa  H_annuus  W_ceracea  Total      prop  \
Orthogroup                                                   
OG0000769          0         0         20     31  0.645161   
OG0005358          0         0          9     14  0.642857   
OG0000208          6         4         38     54  0.703704   
OG0000019         15        20         25    140  0.178571   
OG0000029          4         3         49    116  0.422414   
OG0000111          3         2         28     71  0.394366   
OG0000647          0         0         28     33  0.848485   
OG0003049          1         1         13     18  0.722222   
OG0001077          2         0         18     28  0.642857   
OG0000724          3         2         21     32  0.656250   

                                                         info  
Orthogroup                                                     
OG0000769                       transcription factor possibly  
OG0005358                       transcription factor possibly  
OG0000208                                      Protien Kinase  
OG0000019                                      Protien Kinase  
OG0000029                                             Disease  
OG0000111                                             Disease  
OG0000647                                             Disease  
OG0003049                               Disease - antibiotic?  
OG0001077                               Disease - antibiotic?  
OG0000724   Calcium/lipid-binding domain, CaLB: salt and d...

In [48]:
topics = [TFs, pk, dis, cal]
for x in topics:
    print(x["info"].unique())
    print(ortho_species_df.loc[x.index]["A_thaliana"])

['transcription factor possibly']
Orthogroup
OG0000769            NaN
OG0005358    AT4G33280.3
Name: A_thaliana, dtype: object
['Protien Kinase']
Orthogroup
OG0000208    NaN
OG0000019    NaN
Name: A_thaliana, dtype: object
['Disease' 'Disease - antibiotic?']
Orthogroup
OG0000029            NaN
OG0000111            NaN
OG0000647    AT5G36930.2
OG0003049            NaN
OG0001077            NaN
Name: A_thaliana, dtype: object
['Calcium/lipid-binding domain, CaLB: salt and drought tolerance']
Orthogroup
OG0000724    AT4G01200.1
Name: A_thaliana, dtype: object


So from looking at arabidopsis results, we have that:
- OG0005358 (TFs) contains AT4G33280.3 (https://gbrowseaws.arabidopsis.org/servlets/TairObject?type=gene&id=1000691561) which is "AP2/B3-like transcriptional factor family protein"
- OG0000647 (Dis) contains AT5G36930.2 (https://gbrowseaws.arabidopsis.org/servlets/TairObject?id=133733&type=locus) which is "Disease resistance protein (TIR-NBS-LRR class) family"
    - description of what TIR-NBS-LRR means (a motif homologous to the cytoplasmic domains of the Drosophila Toll protein and the mammalian interleukin-1 receptor (TIR), nucleotide-binding site (NBS), a leucine-rich repeat (LRR)) (https://www.ncbi.nlm.nih.gov/pmc/articles/PMC3472223/#__sec4title).
    - So it is about sequence homology rather than necessarily what the gene does - R gene
- OG0000724 (Cal) contains AT4G01200.1 (https://www.arabidopsis.org/servlets/TairObject?type=locus&name=AT4G01200) which is "Calcium-dependent lipid-binding (CaLB domain) family protein" as specified. it is apparently expressed in guard cells, but TAIR does not have records for the biological process. This other CaLB gene (https://www.arabidopsis.org/servlets/TairObject?type=locus&name=AT3G61050) is the one identified to be involved in salt and drought tollerance https://academic.oup.com/jxb/article/62/8/2679/478984
    - in that paper they talk abou the importance of C2 domains (which our unknown one does have) in stress responses generally "Eperiments with *Atclb* T-DNA knockout lines revealed that these mutants can withstand prolonged drought and salt stress, suggesting involvement of the *Atclb* gene in stress signalling as a negative regulator. Similarly, Yang *et al.* (2006) showed that *Arabidopsis* BAP1 protein with an evolutionarily conserved C2 domain is a negative regulator of defence responses."

In [49]:
ortho_species_df.loc[dis.index]

A_thaliana                                     S_lycopersicum  \
Orthogroup                                                                   
OG0000029           NaN  Solyc02g036270.2.1, Solyc03g094100.1.1, Solyc0...   
OG0000111           NaN  Solyc01g102840.2.1, Solyc01g102850.1.1, Solyc0...   
OG0000647   AT5G36930.2                                 Solyc09g007710.2.1   
OG0003049           NaN                                                NaN   
OG0001077           NaN  Solyc07g043020.1.1, Solyc07g043030.1.1, Solyc0...   

                                                  S_tuberosum  \
Orthogroup                                                      
OG0000029   PGSC0003DMP400001764, PGSC0003DMP400005107, PG...   
OG0000111   PGSC0003DMP400005063, PGSC0003DMP400022408, PG...   
OG0000647                                PGSC0003DMP400048338   
OG0003049                                PGSC0003DMP400017151   
OG0001077          PGSC0003DMP400023996, PGSC0003DMP400057770   

                                                  A_chinensis  \
Orthogroup                                                      
OG0000029   CEY00_Acc03815, CEY00_Acc03888, CEY00_Acc03889...   
OG0000111   CEY00_Acc05254, CEY00_Acc06660, CEY00_Acc06661...   
OG0000647                      CEY00_Acc00750, CEY00_Acc12912   
OG0003049                      CEY00_Acc14622, CEY00_Acc14623   
OG0001077                      CEY00_Acc21775, CEY00_Acc21776   

                            D_carota  \
Orthogroup                             
OG0000029                        NaN   
OG0000111   DCAR_022320, DCAR_029849   
OG0000647                        NaN   
OG0003049                        NaN   
OG0001077                        NaN   

                                                     L_sativa  \
Orthogroup                                                      
OG0000029   Lsat_1_v5_gn_0_10760.1, Lsat_1_v5_gn_1_83140.1...   
OG0000111   Lsat_1_v5_gn_5_101660.1, Lsat_1_v5_gn_5_101700...   
OG0000647                                                 NaN   
OG0003049                               Lsat_1_v5_gn_9_8641.1   
OG0001077    Lsat_1_v5_gn_5_138141.1, Lsat_1_v5_gn_9_116901.1   

                                                     H_annuus  \
Orthogroup                                                      
OG0000029   HanXRQChr01g0017431, HanXRQChr09g0260361, HanX...   
OG0000111            HanXRQChr10g0293411, HanXRQChr10g0298171   
OG0000647                                                 NaN   
OG0003049                                 HanXRQChr10g0281011   
OG0001077                                                 NaN   

                                                    W_ceracea  
Orthogroup                                                     
OG0000029   file_1_file_1_g11039.t1, file_1_file_1_g11040....  
OG0000111   file_1_file_1_g34922.t1, file_1_file_1_g34925....  
OG0000647   file_1_file_1_g17160.t1, file_1_file_1_g17160....  
OG0003049   file_1_file_1_g4143.t1, file_1_file_1_g44729.t...  
OG0001077   file_1_file_1_g27696.t1, file_1_file_1_g27697....

Things I want to write out so I can blast these results:

In [50]:
disease = ortho_species_df.loc[dis.index[0]]["W_ceracea"].split(", ") + \
ortho_species_df.loc[dis.index[1]]["W_ceracea"].split(", ") + \
ortho_species_df.loc[dis.index[3]]["W_ceracea"].split(", ") + \
ortho_species_df.loc[dis.index[4]]["W_ceracea"].split(", ") # + \
#ortho_species_df.loc[dis.index[2]]["W_ceracea"].split(", ")

In [51]:
protien_k = ortho_species_df.loc[pk.index[0]]["W_ceracea"].split(", ") + ortho_species_df.loc[pk.index[1]]["W_ceracea"].split(", ")

In [52]:
trans_f = ortho_species_df.loc[TFs.index[0]]["W_ceracea"].split(", ")

In [53]:
#with open("/home/jemimah/analysis/v3/orthofinder/20200911/interesting_protein_lists/disease.list", "w") as f:
#    f.write("\n".join(disease) + "\n")
#with open("/home/jemimah/analysis/v3/orthofinder/20200911/interesting_protein_lists/protien_k.list", "w") as f:
#    f.write("\n".join(protien_k) + "\n")
#with open("/home/jemimah/analysis/v3/orthofinder/20200911/interesting_protein_lists/trans_f.list", "w") as f:
#    f.write("\n".join(trans_f) + "\n")

In [54]:
#getting the disease results for each orthogroup:
dis_OG0000029 = ortho_species_df.loc[dis.index[0]]["W_ceracea"].split(", ")
dis_OG0000111 = ortho_species_df.loc[dis.index[1]]["W_ceracea"].split(", ")
dis_OG0000647 = ortho_species_df.loc[dis.index[2]]["W_ceracea"].split(", ")
with open("/home/jemimah/analysis/v3/orthofinder/20200911/interesting_protein_lists/dis_OG0000029.list", "w") as f:
    f.write("\n".join(dis_OG0000029) + "\n")
with open("/home/jemimah/analysis/v3/orthofinder/20200911/interesting_protein_lists/dis_OG0000111.list", "w") as f:
    f.write("\n".join(dis_OG0000111) + "\n")
with open("/home/jemimah/analysis/v3/orthofinder/20200911/interesting_protein_lists/dis_OG0000647.list", "w") as f:
    f.write("\n".join(dis_OG0000647) + "\n")

In [55]:
%%bash
cd /home/jemimah/analysis/v3/orthofinder/20200911/interesting_protein_lists/
#for x in *.list; do seqtk subseq ~/analysis/v3/braker2/20200709_real/braker2.codingseq ${x} > ${x%%.list}.fa; done
#for x in dis_*.fa; do grep -v ">" ${x} > ${x%%.fa}_merged.fa; done
wc -l *.list
grep -c ">" *.fa

 108 disease.list
  49 dis_OG0000029.list
  28 dis_OG0000111.list
  28 dis_OG0000647.list
  63 protien_k.list
  20 trans_f.list
 296 total
disease.fa:108
disease_merged.fa:0
dis_OG0000029.fa:49
dis_OG0000029_merged.fa:0
dis_OG0000111.fa:28
dis_OG0000111_merged.fa:0
dis_OG0000647.fa:28
dis_OG0000647_merged.fa:0
protien_k.fa:63
protien_k_merged.fa:0
trans_f.fa:20
trans_f_merged.fa:0


## Topics Summary of above

In [56]:
topics = [TFs, pk, dis, cal]
topics_ogs = {}
for x in topics:
    topics_ogs[x["info"][0]] = list(x.index)

#### Transcription Factors

In [57]:
print(TFs["info"].unique()[0])
ips_summary(list(TFs.index), "table")
TFs

transcription factor possibly
OG0000769


OG0005358


A_thaliana  S_lycopersicum  S_tuberosum  A_chinensis  D_carota  \
Orthogroup                                                                   
OG0000769            0               3            2            1         5   
OG0005358            1               1            1            2         0   

            L_sativa  H_annuus  W_ceracea  Total      prop  \
Orthogroup                                                   
OG0000769          0         0         20     31  0.645161   
OG0005358          0         0          9     14  0.642857   

                                     info  
Orthogroup                                 
OG0000769   transcription factor possibly  
OG0005358   transcription factor possibly

In [67]:
for x in TFs.columns[:-4]:
    print(x)
    print(ortho_species_df.loc[TFs.index][x])
    print(list(ortho_species_df.loc[TFs.index][x]))

A_thaliana
Orthogroup
OG0000769            NaN
OG0005358    AT4G33280.3
Name: A_thaliana, dtype: object
[nan, 'AT4G33280.3']
S_lycopersicum
Orthogroup
OG0000769    Solyc07g007070.1.1, Solyc07g007080.1.1, Solyc0...
OG0005358                                   Solyc02g090170.2.1
Name: S_lycopersicum, dtype: object
['Solyc07g007070.1.1, Solyc07g007080.1.1, Solyc07g007100.1.1', 'Solyc02g090170.2.1']
S_tuberosum
Orthogroup
OG0000769    PGSC0003DMP400053545, PGSC0003DMP400053553
OG0005358                          PGSC0003DMP400017857
Name: S_tuberosum, dtype: object
['PGSC0003DMP400053545, PGSC0003DMP400053553', 'PGSC0003DMP400017857']
A_chinensis
Orthogroup
OG0000769                    CEY00_Acc00677
OG0005358    CEY00_Acc04681, CEY00_Acc23701
Name: A_chinensis, dtype: object
['CEY00_Acc00677', 'CEY00_Acc04681, CEY00_Acc23701']
D_carota
Orthogroup
OG0000769    DCAR_014925, DCAR_014926, DCAR_014927, DCAR_01...
OG0005358                                                  NaN
Name: D_carota, dtyp

**OG0000769**   
Walli genes: 20   
All interpro results: 49   
Unique interpro results: 3   
- HOMEODOMAIN PROTEIN 8, PUTATIVE-RELATED (http://www.pantherdb.org/panther/family.do?clsAccession=PTHR35121:SF4)
- probably a transcription factor https://www.sciencedirect.com/science/article/pii/096800049290434B?via%3Dihub
- consensus disorder prediction also present in TFs

other protiens in this panther family for arabidopsis are uncharacterised and dissordered.  

because I can't find anything of interest for this set, and even find out what panther means by "homeodomain protien 8 " (there are a lot of homeodomain protiens???) I am dropping it.

**OG0005358**  
Walli genes: 9   
All interpro results: 116   
Unique interpro results: 11   
- Panther is the same as the ProSiteProfiles one below it
- Seems to be a DNA binding something: B3 is exclusively TFs https://en.wikipedia.org/wiki/B3_domain  
  
    
- OG0005358 (TFs) contains AT4G33280.3 (https://gbrowseaws.arabidopsis.org/servlets/TairObject?type=gene&id=1000691561) which is "AP2/B3-like transcriptional factor family protein"
- the interproscan results for this gene also have the barrel thing http://www.ebi.ac.uk/interpro/search/text/A0A1P8B3A7/#table

[plant specific???]

blast for trans_f.fa reports no significant matches for highly simillar.  
at less simllar, mostly uncharacterised sequence matches.

#### Protien Kinases

In [59]:
print(pk["info"].unique()[0])
ips_summary(list(pk.index), "table")
pk

Protien Kinase
OG0000208


OG0000019


A_thaliana  S_lycopersicum  S_tuberosum  A_chinensis  D_carota  \
Orthogroup                                                                   
OG0000208            0               1            1            2         2   
OG0000019            0              10           36           12        22   

            L_sativa  H_annuus  W_ceracea  Total      prop            info  
Orthogroup                                                                  
OG0000208          6         4         38     54  0.703704  Protien Kinase  
OG0000019         15        20         25    140  0.178571  Protien Kinase

In [60]:
for x in pk.columns[:-4]:
    print(x)
    print(ortho_species_df.loc[pk.index][x])
    print(list(ortho_species_df.loc[pk.index][x]))

A_thaliana
Orthogroup
OG0000208    NaN
OG0000019    NaN
Name: A_thaliana, dtype: object
[nan, nan]
S_lycopersicum
Orthogroup
OG0000208                                   Solyc04g078410.2.1
OG0000019    Solyc02g079610.2.1, Solyc02g081470.2.1, Solyc0...
Name: S_lycopersicum, dtype: object
['Solyc04g078410.2.1', 'Solyc02g079610.2.1, Solyc02g081470.2.1, Solyc02g081510.1.1, Solyc02g086210.2.1, Solyc03g007210.2.1, Solyc04g007380.1.1, Solyc05g009040.2.1, Solyc05g009050.2.1, Solyc05g009090.1.1, Solyc12g040370.1.1']
S_tuberosum
Orthogroup
OG0000208                                 PGSC0003DMP400005946
OG0000019    PGSC0003DMP400008038, PGSC0003DMP400010170, PG...
Name: S_tuberosum, dtype: object
['PGSC0003DMP400005946', 'PGSC0003DMP400008038, PGSC0003DMP400010170, PGSC0003DMP400010663, PGSC0003DMP400016043, PGSC0003DMP400021362, PGSC0003DMP400021364, PGSC0003DMP400021365, PGSC0003DMP400024484, PGSC0003DMP400024750, PGSC0003DMP400024751, PGSC0003DMP400024752, PGSC0003DMP400024753, PGSC0003DMP40002

**OG0000208** - Protein Kinases  
Walli genes: 38   
All interpro results: 571   
Unique interpro results: 38  
- Panther is also Protein Kinases

~**OG0000019**~ -  Protein kinases, maybe rust resistent?
Walli genes: 25   
All interpro results: 244   
Unique interpro results: 20  
- http://www.pantherdb.org/panther/family.do?clsAccession=PTHR27009: RUST RESISTANCE KINASE LR10-RELATED
    - cannot find how it is rust resistant??
    
For both, all Tomato genes are Serine/threonine protein kinase (based off their interproscan results)(eg.http://223.31.159.9/tomato2/getGene.php?trans_fac_id=Solyc04g078410.2.1) except for one (http://223.31.159.9/tomato2/getGene.php?trans_fac_id=Solyc12g040370.1.1) which is "core"

Blast online alignments for protien_k_merged.fa (looking at specific genes with protien_k.fa):
- predicted putative receptor protein kinase ZmPK1 in a range of species (ZmPK1 refferes to the maize Protien Kinase 1 described here: https://www.nature.com/articles/345743a0)
- The second half has far more rust resistent ones - LR10-like (Paper for LR10 - https://pubmed.ncbi.nlm.nih.gov/9700067/)

Both ZmPK1 (https://www.uniprot.org/uniprot/P17801) and LR10 (https://www.uniprot.org/uniprot/P93604) are Serine/threonine protein kinases.  
Both are thought to be probable receptors  


Diseases

In [61]:
print(dis["info"].unique()[0])
ips_summary(list(dis.index), "table")
dis

Disease
OG0000029


OG0000111


OG0000647


OG0003049


OG0001077


A_thaliana  S_lycopersicum  S_tuberosum  A_chinensis  D_carota  \
Orthogroup                                                                   
OG0000029            0              18           26           16         0   
OG0000111            0               8           10           18         2   
OG0000647            1               1            1            2         0   
OG0003049            0               0            1            2         0   
OG0001077            0               4            2            2         0   

            L_sativa  H_annuus  W_ceracea  Total      prop  \
Orthogroup                                                   
OG0000029          4         3         49    116  0.422414   
OG0000111          3         2         28     71  0.394366   
OG0000647          0         0         28     33  0.848485   
OG0003049          1         1         13     18  0.722222   
OG0001077          2         0         18     28  0.642857   

                             info  
Orthogroup                         
OG0000029                 Disease  
OG0000111                 Disease  
OG0000647                 Disease  
OG0003049   Disease - antibiotic?  
OG0001077   Disease - antibiotic?

In [62]:
set(list(ortho_species_df.loc[["OG0000111"]]["W_ceracea"])) & set(list(ortho_species_df.loc[["OG0000647"]]["W_ceracea"]))
#set(list(ortho_species_df.loc[["OG0000111"]]["W_ceracea"])) & set(list(ortho_species_df.loc[["OG0000029"]]["W_ceracea"]))
#set(list(ortho_species_df.loc[["OG0000029"]]["W_ceracea"])) & set(list(ortho_species_df.loc[["OG0000647"]]["W_ceracea"]))

set()

In [63]:
print("OG0000029")
for x in pk.columns[:-4]:
    print(x)
    print(ortho_species_df.loc["OG0000029"][x])
print("OG0000111")
for x in pk.columns[:-4]:
    print(x)
    print(ortho_species_df.loc["OG0000111"][x])

OG0000029
A_thaliana
nan
S_lycopersicum
Solyc02g036270.2.1, Solyc03g094100.1.1, Solyc04g005540.2.1, Solyc04g005550.1.1, Solyc04g048920.1.1, Solyc07g044800.2.1, Solyc10g054940.1.1, Solyc10g054970.1.1, Solyc10g054990.1.1, Solyc10g055050.1.1, Solyc10g055110.1.1, Solyc10g055140.1.1, Solyc10g055170.1.1, Solyc12g005530.1.1, Solyc12g016220.1.1, Solyc12g044180.1.1, Solyc12g044190.1.1, Solyc12g044200.1.1
S_tuberosum
PGSC0003DMP400001764, PGSC0003DMP400005107, PGSC0003DMP400005435, PGSC0003DMP400013406, PGSC0003DMP400014641, PGSC0003DMP400019233, PGSC0003DMP400019235, PGSC0003DMP400019349, PGSC0003DMP400019350, PGSC0003DMP400020238, PGSC0003DMP400020241, PGSC0003DMP400020244, PGSC0003DMP400023299, PGSC0003DMP400026426, PGSC0003DMP400029390, PGSC0003DMP400029592, PGSC0003DMP400030444, PGSC0003DMP400030446, PGSC0003DMP400032936, PGSC0003DMP400037237, PGSC0003DMP400038006, PGSC0003DMP400038184, PGSC0003DMP400048746, PGSC0003DMP400048747, PGSC0003DMP400048785, PGSC0003DMP400065504
A_chinensis
CEY00_

**OG0000029** - disease resistance  
Walli genes: 49   
All interpro results: 391   
Unique interpro results: 21  
- Prints result is for disease resistence (hard to find more info on, though)
- PTHR33463:SF21 can't be found??? but PTHR33463 is a family with 973 genes, some of the subfamilies being disease resistence http://www.pantherdb.org/list/list.do?numPerPage=100&save=yes&searchModType=numperpage&listType=6

Tomato  
- Solyc02g036270.2.1 desease resistance: NB-LRR family (https://string-db.org/network/4081.Solyc02g036270.2.1, https://www.ncbi.nlm.nih.gov/pmc/articles/PMC5832704/)
- generally can be found: http://223.31.159.9/tomato2/r-genes.html

**OG0000111** - Maybe disease resistance?   
Walli genes: 28    
All interpro results: 460   
Unique interpro results: 23  
- Seems to be a mixture of results I'vve already had?
    - PRINTS reports desiease resistence
    - Gene3D. Panther, others report Leucine-rich repeat domain
    - Toll/interleukin results seem in plant cytoplasmic host defence https://www.ebi.ac.uk/interpro/entry/InterPro/IPR000157/)
  
- Punitative tomato r genes in matches:
    - http://223.31.159.9/tomato2/getGene.php?trans_fac_id=Solyc02g032650.2.1; http://223.31.159.9/tomato2/getGene.php?trans_fac_id=Solyc02g032230.1.1; http://223.31.159.9/tomato2/getGene.php?trans_fac_id=Solyc02g032230.1.1 ; http://prgdb.org/prgdb/genes/type/putative/2135094; http://www.prgdb.org/prgdb/genes/type/putative/2135035
  
- combine with the next one in discussion?
    - Different orthogroup match rates
    - different Walli genes
    - :. different "origin"
    - same domains :. same function?

**OG0000647** - very simillar to the one above   
Walli genes: 28   
All interpro results: 388   
Unique interpro results: 27  
- might actuaklly be the same results (with different counts)
- if they're so similalr, why are they two different orthogroups?  
  
  
- OG0000647 (Dis) contains AT5G36930.2 (https://gbrowseaws.arabidopsis.org/servlets/TairObject?id=133733&type=locus) which is "Disease resistance protein (TIR-NBS-LRR class) family"
    - description of what TIR-NBS-LRR means (a motif homologous to the cytoplasmic domains of the Drosophila Toll protein and the mammalian interleukin-1 receptor (TIR), nucleotide-binding site (NBS), a leucine-rich repeat (LRR)) (https://www.ncbi.nlm.nih.gov/pmc/articles/PMC3472223/#__sec4title).
    - So it is about sequence homology rather than necessarily what the gene does - R gene
    - it is a type D (https://link.springer.com/article/10.1007%2Fs11103-008-9293-9) which means that it's last residue is D (Aspartate)
    - we dont know what they do. https://genomebiology.biomedcentral.com/articles/10.1186/gb-2006-7-4-212, and there are usually giant families of them in plants.
    
    
~**OG0003049**~  
Walli genes: 13   
All interpro results: 79   
Unique interpro results: 9   
- Gene3D has it as an antibiotic? http://www.cathdb.info/version/v4_2_0/superfamily/2.60.120.330
- but all the rest seem to just be 2OG-Fe(II) oxygenase superfamily	- apparently the families overlap? http://www.ebi.ac.uk/interpro/entry/InterPro/IPR026992/
- antibiotic is too complicated

~**OG0001077**~ - Glycosyltransferase/glycogen phosphorylase - anti-biotics or pigment?  
Walli genes: 18  
All interpro results: 117  
Unique interpro results: 6  

- cdd link https://www.ncbi.nlm.nih.gov/Structure/cdd/cd03784 - involved in the final stages of the biosynthesis of antibiotics

- interproscan codes have it for plants as anthocyanin pigment biosynthesis (https://www.ebi.ac.uk/interpro/entry/InterPro/IPR002213/, https://www.ebi.ac.uk/interpro/entry/InterPro/IPR035595/)

Three orthogroups likely disease resistance:
1.	Unknown disease resistance (OG0000029)
    - 49 Walli, present in all else except Arabidopsis and carrot (3-26)
    - Matches Tomato genes with known? Gene resistance through exact properties unknown
        -	Solyc02g036270.2.1 desease resistance: NB-LRR family (https://string-db.org/network/4081.Solyc02g036270.2.1, https://www.ncbi.nlm.nih.gov/pmc/articles/PMC5832704/)
        -	NBS-LRR and NB-LRR are the same?
        - Blast results have “like” these Arabidopsis genes: https://www.uniprot.org/uniprot/Q9T048, https://www.arabidopsis.org/servlets/TairObject?name=AT1G12280&type=locus (likely to be a bad match, because a different kind of resistance gene with interproscan results mine don’t have), https://www.arabidopsis.org/servlets/TairObject?accession=locus:2131689 
    - Compared to next one, has different species in the orthogroup and different walli genes (difference of orthology?) but same domains (same function?)
2.	Unsorted disease resistence (OG0000111)
    -	28 Walli genes, present in all except a thaliana (2-18)
    -	Matches tomato genes with punitative r gene roles
    -	Very simillar to the next one
3.	TIR-NBS-LRR class (OG0000647)
    -	28 Walli genes, doesn’t match carrot, lettuce, sunflower. (1-2 matches for others)
    -	Arabidopsis match is declared disease resistance (is this punitative too?). 
    -	Generally, we don’t know how R genes work and are combining homology with testing a few. 



#### Calcium

In [64]:
print(cal["info"].unique()[0])
ips_summary(list(cal.index), "table")
cal

Calcium/lipid-binding domain, CaLB: salt and drought tolerance
OG0000724


A_thaliana  S_lycopersicum  S_tuberosum  A_chinensis  D_carota  \
Orthogroup                                                                   
OG0000724            1               1            1            2         1   

            L_sativa  H_annuus  W_ceracea  Total     prop  \
Orthogroup                                                  
OG0000724          3         2         21     32  0.65625   

                                                         info  
Orthogroup                                                     
OG0000724   Calcium/lipid-binding domain, CaLB: salt and d...

In [65]:
for x in cal.columns[:-4]:
    print(x)
    print(ortho_species_df.loc["OG0000724"][x])

A_thaliana
AT4G01200.1
S_lycopersicum
Solyc01g095360.2.1
S_tuberosum
PGSC0003DMP400000188
A_chinensis
CEY00_Acc07593, CEY00_Acc15949
D_carota
DCAR_025234
L_sativa
Lsat_1_v5_gn_0_42241.1, Lsat_1_v5_gn_4_34140.1, Lsat_1_v5_gn_5_79441.1
H_annuus
HanXRQChr16g0523831, HanXRQChr17g0561051


**OG0000724** - Calcium/lipid-binding domain, CaLB  
Walli genes: 21   
All interpro results: 147   
Unique interpro results: 10   
- https://www.ncbi.nlm.nih.gov/gene?cmd=DetailsSearch&term=AT3G61050: "It can bind ceramide and is involved in drought and salt tolerance."

- OG0000724 (Cal) contains AT4G01200.1 (https://www.arabidopsis.org/servlets/TairObject?type=locus&name=AT4G01200) which is "Calcium-dependent lipid-binding (CaLB domain) family protein" as specified. it is apparently expressed in guard cells, but TAIR does not have records for the biological process. This other CaLB gene (https://www.arabidopsis.org/servlets/TairObject?type=locus&name=AT3G61050) is the one identified to be involved in salt and drought tollerance https://academic.oup.com/jxb/article/62/8/2679/478984
    - in that paper they talk abou the importance of C2 domains (which our unknown one does have) in stress responses generally "Eperiments with *Atclb* T-DNA knockout lines revealed that these mutants can withstand prolonged drought and salt stress, suggesting involvement of the *Atclb* gene in stress signalling as a negative regulator. Similarly, Yang *et al.* (2006) showed that *Arabidopsis* BAP1 protein with an evolutionarily conserved C2 domain is a negative regulator of defence responses."

In [66]:
keeps = ["OG0005358", "OG0000029", "OG0000111", "OG0000647", "OG0000208", "OG0000019", "OG0000724"]
keeps_df = TFs.append(pk).append(dis).append(cal)
keeps_df = keeps_df[keeps_df.index.isin(keeps)].reset_index()
paper_headers = list(keeps_df.columns[:-2].append(keeps_df.columns[-1:]))
keeps_df = keeps_df[paper_headers]
keeps_df

Orthogroup  A_thaliana  S_lycopersicum  S_tuberosum  A_chinensis  D_carota  \
0  OG0005358           1               1            1            2         0   
1  OG0000208           0               1            1            2         2   
2  OG0000019           0              10           36           12        22   
3  OG0000029           0              18           26           16         0   
4  OG0000111           0               8           10           18         2   
5  OG0000647           1               1            1            2         0   
6  OG0000724           1               1            1            2         1   

   L_sativa  H_annuus  W_ceracea  Total  \
0         0         0          9     14   
1         6         4         38     54   
2        15        20         25    140   
3         4         3         49    116   
4         3         2         28     71   
5         0         0         28     33   
6         3         2         21     32   

                                                info  
0                      transcription factor possibly  
1                                     Protien Kinase  
2                                     Protien Kinase  
3                                            Disease  
4                                            Disease  
5                                            Disease  
6  Calcium/lipid-binding domain, CaLB: salt and d...